src/libvideo/YouTube.cs

```diff
@@ -58,48 +58,53 @@ private IEnumerable<YouTubeVideo> ParseVideos(string source)
             string jsPlayer = "http:" + Json.GetKey("js", source).Replace(@"\/", "/");
 
             string map = Json.GetKey("url_encoded_fmt_stream_map", source);
-            var queries = map.Split(',').Select(Unscramble);
-
+            var queries = Unscramble(map.Split(','));
+            
             foreach (var query in queries)
                 yield return new YouTubeVideo(title, query, jsPlayer);
 
             string adaptiveMap = Json.GetKey("adaptive_fmts", source);
 
-            queries = adaptiveMap.Split(',').Select(Unscramble);
+            queries = Unscramble(adaptiveMap.Split(','));
 
             foreach (var query in queries)
                 yield return new YouTubeVideo(title, query, jsPlayer);
         }
 
         // TODO: Consider making this static...
-        private UnscrambledQuery Unscramble(string queryString)
+        private IEnumerable<UnscrambledQuery> Unscramble(string[] queryStrings)
         {
-            queryString = queryString.Replace(@"\u0026", "&");
-            var query = new Query(queryString);
-            string uri = query["url"];
-
-            bool encrypted = false;
-            string signature;
-
-            if (query.TryGetValue("s", out signature))
+            foreach (var queryString in queryStrings)
             {
-                encrypted = true;
-                uri += GetSignatureAndHost(signature, query);
-            }
-            else if (query.TryGetValue("sig", out signature))
-                uri += GetSignatureAndHost(signature, query);
+                var query = new Query(queryString.Replace(@"\u0026", "&"));
+                if (query.ContainsKey("url"))
```

Comment:
Github makes this PR look messier than it actually is; really the only substantive change is the addition of this if statement. I also changed the method signature to return the IEnumerable so it's called once instead of multiple times. Everything else is just indentation adjustment.